default-fields/0000-default-fields.md

```diff
@@ -6,31 +6,285 @@
 # Summary
 [summary]: #summary
 
-One para explanation of the feature.
+Allow `struct` definitions to supply default values for individual fields, and then allow those fields to be omitted from `struct` initialisation:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32,
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+};
+```
 
 # Motivation
 [motivation]: #motivation
 
-Why are we doing this? What use cases does it support? What is the expected outcome?
+Rust allows you to create an instance of a `struct` using a literal syntax. This requires that all fields in the `struct` are assigned a value, so it can be inconvenient for large `struct`s whose fields usually receive the same values. Literals also can't be used to initialise `struct`s with innaccessible private fields. Functional record updates can reduce noise when a `struct` derives `Default`, but are also invalid when the `struct` has private fields.
+
+To work around these shortcomings, you can create constructor functions:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool,
+    c: i32,
+}
+
+impl Foo {
+    // Constructor function.
+    fn new(a: &'static str, c: i32) -> Self {
+        Foo {
+            a: a,
+            b: true,
+            c: c
+        }
+    }
+}
+
+let foo = Foo::new("Hello", 42);
+```
+
+The problem with a constructor is that you need one for each combination of fields a caller can supply. To work around this, you can use builders, like [`process::Command`](https://doc.rust-lang.org/stable/std/process/struct.Command.html) in the standard library. Builders enable more advanced initialisation, but need additional boilerplate.
+
+This RFC proposes a solution to improve `struct` literal ergonomics so they can be used for `struct`s with private fields and to reduce initialisation boilerplate for simple scenarios. This is achieved by letting callers omit fields from initialisation when a default is specified for that field. This syntax also allows allows fields to be added to `struct`s in a backwards compatible way, by providing defaults for new fields.
+
+Field defaults allow a caller to initialise a `struct` with default values without needing builders or a constructor function:
+
+```rust
+struct Foo {
+    a: &'static str,
+    b: bool = true,
+    c: i32,
+}
+
+let foo = Foo {
+    a: "Hello",
+    c: 42,
+};
```

Comment:
I think it would be interesting for this example to give defaults for every field and then show a few different struct lits supplying different combinations of fields.